src/axom/slic/core/LogStream.hpp

```diff
@@ -144,7 +148,40 @@ class LogStream
    */
   virtual void push() {};
 
+  /*!
+   * \brief Determines whether it is possible to reliably check if pending messages exist
+   *
+   * \return Returns True when it is possible to reliably check whether pending messages exist
+   */
+  virtual bool canHavePendingMessages() { return false; };
```

Comment:
In my opinion this is kind of awkward. I think its better to just have the `hasPendingMesages` and return `false` for any that will not hold onto messages by their design.